build-infrastructure/minimal-signing-build-stack.yml

```diff
@@ -0,0 +1,233 @@
+---
+AWSTemplateFormatVersion: '2010-09-09'
+Description: A template that creates a minimal signing CodeBuild environment and stores it in ECR
+
+Parameters:
+  SignerImageRepositoryName:
+    Type: String
+    Description: the name of the ECR repository to upload the signer image to
+    Default: ecs-agent-signer
+  RepositoryImageRententionPeriodInDays:
+    Type: Number
+    Default: 180
+  ImageCodeBuildProjectName:
+    Type: String
+    Description: the name of the CodeBuild project that builds the signing Docker image
+    Default: ecs-agent-signer-image-build
+  DockerBuildLogsGroupName:
+    Type: String
+    Description: The name of the log group to store the signing Docker image logs in
+    Default: signer-image-build-logs
+  CodeStarConnectionArn:
+    Type: String
+    Description: The ARN of the connection to use to connect to GitHub
+  GitHubRepositoryUrl:
+    Type: String
+    Description: The repository to pull the Dockerfile from so that we can build the signer image
+    Default: https://github.com/aws/amazon-ecs-agent
+  GitHubBranchName:
+    Type: String
+    Description: The branch to use from the repository mentioned above
+    Default: master
+  ImageBuildFrequencyCronExpression:
+    Type: String
+    Description: A cron expression to periodically build the signing image, can be left blank to disable periodic builds. See https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html
+    Default: 30 10 ? * 4 *
+  PeriodicBuildTriggerName:
+    Type: String
+    Description: The name of the CloudWatch Event rule, ignored if ImageBuildFrequencyCronExpression is blank
+    Default: PeriodicallyTriggerSigningImageBuild
```

Comment:
Can you talk about why you chose a periodic build trigger rather than an event-based trigger?  How often do we expect that a new image should be built -- seems the default will be once per week?  Is the frequency based on Amazon Linux (base image) updates?

Reply:
I actually tried to figure out a good event based trigger (listing the ECR AL2 image tags and figuring out if something is new) but I couldn't figure out a good event based trigger. But I did want to continue to rebuild this image because I didn't want the docker image to become stale, thus opening us up for unaddressed fixes that do exist in newer versions of the base image. 

I tried to be quicker than the smallest AL2 release cadence that I found recently so I found that it was about a week. 

The idea here was to get something out that does automatically build and we don't have to take on that ops burden too much to cater to our infrastructure. And I'm hoping that as we get a better understanding of a good event-based trigger, we can swap that in for a schedule based trigger.